corehq/form_processor/submission_post.py

```diff
@@ -290,21 +308,18 @@ def process_signals(self, instance):
                 ) % (func, instance.form_id, type(resp).__name__, error_message))
                 errors.append(error_message)
         if errors:
-            self.interface.xformerror_from_xform_instance(instance, ", ".join(errors))
+            self.interface.xformerror_from_xform_instance(instance, ", ".join(errors), with_new_id=True)
             self.formdb.update_form_problem_and_state(instance)
```

Comment:
This function will have to change since it doesn't support changing the form ID. For couch this isn't too much of an issue since we just save the form but for SQL we'll have to delete the old form and save the new form since it might move to a new shard.